lighthouse-core/test/lib/network-request-test.js

```diff
@@ -6,189 +6,170 @@
-        proxyEnd: -1,
-        dnsStart: -1,
-        dnsEnd: -1,
-        connectStart: -1,
-        connectEnd: -1,
-        sslStart: -1,
-        sslEnd: -1,
-        workerStart: -1,
-        workerReady: -1,
-        sendStart: -1,
-        sendEnd: -1,
-        pushStart: -1,
-        pushEnd: -1,
-        receiveHeadersEnd: -1,
+      return {
+        // units = seconds
+        startTime: 0,
+        endTime: 2,
+        responseReceivedTime: 1,
+
+        // units = ms
+        responseHeaders: [
+          {name: NetworkRequest.HEADER_TOTAL, value: '10000'},
+          {name: NetworkRequest.HEADER_TCP, value: '5000'},
+          {name: NetworkRequest.HEADER_REQ, value: '2500'},
+          {name: NetworkRequest.HEADER_SSL, value: '1000'},
+          {name: NetworkRequest.HEADER_RES, value: '2500'},
+        ],
       };
-
-      // units = ms
-      req.responseHeaders = [{name: NetworkRequest.HEADER_TOTAL, value: 10000},
-        {name: NetworkRequest.HEADER_TCP, value: 5000},
-        {name: NetworkRequest.HEADER_REQ, value: 2500},
-        {name: NetworkRequest.HEADER_SSL, value: 1000},
-        {name: NetworkRequest.HEADER_RES, value: 2500}];
-      return req;
     }
 
-    it('does nothing if not Lightrider', function() {
+    it('updates lrStatistics if in Lightrider', () => {
       const req = getRequest();
-      global.isLightrider = false;
 
-      assert.equal(req.startTime, 0);
-      assert.equal(req.endTime, 2);
-      assert.equal(req.responseReceivedTime, 1);
-      req._updateTimingsForLightrider();
-      assert.equal(req.startTime, 0);
-      assert.equal(req.endTime, 2);
-      assert.equal(req.responseReceivedTime, 1);
+      const devtoolsLog = networkRecordsToDevtoolsLog([req]);
+      global.isLightrider = true;
+      const record = NetworkRecorder.recordsFromLogs(devtoolsLog)[0];
+
+      expect(record.startTime).toBe(0);
+      expect(record.endTime).toBe(2);
+      expect(record.responseReceivedTime).toBe(1);
+      expect(record.lrStatistics).toStrictEqual({
+        endTimeDeltaMs: -8000,
+        TCPMs: 5000,
+        requestMs: 2500,
+        responseMs: 2500,
+      });
     });
 
-    it('does nothing if no TotalTime', function() {
+    it('does nothing if not Lightrider', () => {
       const req = getRequest();
-      global.isLightrider = true;
-      req.responseHeaders = req.responseHeaders.filter(item => item.name
-        !== NetworkRequest.HEADER_TOTAL);
-
-      assert.equal(req.startTime, 0);
-      assert.equal(req.endTime, 2);
-      assert.equal(req.responseReceivedTime, 1);
-      req._updateTimingsForLightrider();
-      assert.equal(req.startTime, 0);
-      assert.equal(req.endTime, 2);
-      assert.equal(req.responseReceivedTime, 1);
+      req.responseHeaders = [];
+
+      const devtoolsLog = networkRecordsToDevtoolsLog([req]);
+      global.isLightrider = false;
+      const record = NetworkRecorder.recordsFromLogs(devtoolsLog)[0];
+
+      expect(record).toMatchObject(req);
```

Comment:
these tests feel a bit hollow when combined with `toMatchObject` since the only affect the code will ever have right now is mutating `lrStatistics` and we'll never asset anything about that property

should we assert something on `lrStatistics`? or are these tests mostly just for saving time when we do the followup with the real manipulation code?

Reply:
> these tests feel a bit hollow when combined with `toMatchObject` since the only effect the code will ever have right now is mutating `lrStatistics` and we'll never asset anything about that property

Yeah, I think these were intended as counterpoints to tests where the other timing numbers were changing. Those tests were removed when that was dropped from the PR, but these remained. Seems ok to leave in since it's a little hard to tease out a dividing line and the timing changes should come very soon, but yeah, these are mostly just a test of `networkRecordsToDevtoolsLog` right now :)

I can put an explicit check for `lrStatistics` just so it's clear it's expected to not be defined.